### ISSUE TITLE
Python 2 UnicodeDecodeError

### ISSUE THREAD

**Issue by llnz** [DRMacIver/hypothesis#99]
Upgrading from 1.7.2 to 1.8.0 causes UnicodeDecodeError exceptions inside hypothesis.

```python
Traceback (most recent call last):
  File "/path/to/project/.hypothesis/eval_source/hypothesis_temporary_module_c366bf4c478e743e982138b9f3a3d1bb7495341c.py", line 5, in test_invariants
    return f(self, shift)
  File "/path/to/lib/python2.7/site-packages/hypothesis/core.py", line 532, in wrapped_test
    print_example=True
  File "/path/to/lib/python2.7/site-packages/hypothesis/executors/executors.py", line 37, in execute
    return function()
  File "/path/to/lib/python2.7/site-packages/hypothesis/core.py", line 332, in run
    text_version = arg_string(test, args, kwargs)
  File "/path/to/lib/python2.7/site-packages/hypothesis/internal/reflection.py", line 285, in arg_string
    bits.append('%s=%s' % (a, value))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 55: ordinal not in range(128)
```

Works fine under python 3.

I'm guessing here, but one of the args contains an django model with a repr string that is/contains unicode (from a hypothesis created model)

**Comment by DRMacIver** [comment]
Ugh. Right. Yes, that's a pretty plausible guess. I'll try to have a fix for you later today.

**Comment by DRMacIver** [comment]
On further investigation... Note that this is a bug in your repr. It's a common enough one that I'm going to try to work around it, but a plain repr(x) call will fail in Python 2 if x.__repr__ returns something not interpretable as ascii.

**Comment by DRMacIver** [comment]
OK. I've just released Hypothesis 1.8.1 which I believe should fix your problem.

If it doesn't then either this is a bug that really needs fixing in your code or I need you to give me a test case because this fixes everything I can think of in this space.

**Comment by llnz** [comment]
Thanks, works well.

Interesting cause, and you are right about it being common, especially in Django model classes.
